packages/graphql/tests/tck/tck-test-files/issues/1115.test.ts

```diff
@@ -0,0 +1,115 @@
+    let typeDefs: DocumentNode;
+    let neoSchema: Neo4jGraphQL;
+
+    beforeAll(() => {
+        typeDefs = gql`
+            type Parent {
+                children: [Child!]! @relationship(type: "HAS", direction: IN)
+            }
+            type Child {
+                tcId: String @unique
+            }
+
+            extend type Child
+                @auth(
+                    rules: [
+                        { operations: [READ, CREATE, UPDATE, DELETE, CONNECT, DISCONNECT], roles: ["upstream"] }
+                        { operations: [READ], roles: ["downstream"] }
+                    ]
+                )
+        `;
+
+        neoSchema = new Neo4jGraphQL({
+            typeDefs,
+            config: { enableRegex: true },
+        });
+    });
+
+    test("multiple connectOrCreate operations with auth", async () => {
+        const query = gql`
+            mutation UpdateParents {
+                updateParents(
+                    connectOrCreate: {
+                        children: [
+                            { where: { node: { tcId: "123" } }, onCreate: { node: { tcId: "123" } } }
+                            { where: { node: { tcId: "456" } }, onCreate: { node: { tcId: "456" } } }
+                        ]
+                    }
+                ) {
+                    info {
+                        nodesCreated
+                    }
+                }
+            }
+        `;
+
+        const req = createJwtRequest("secret", {});
+        const result = await translateQuery(neoSchema, query, {
+            req,
+        });
+
+        expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
+            "MATCH (this:Parent)
+            	WITH this
+            CALL {
+            	WITH this
+            	CALL apoc.util.validate(NOT(ANY(r IN [\\"upstream\\"] WHERE ANY(rr IN $auth.roles WHERE r = rr))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
+            MERGE (this_connectOrCreate_children_this1:\`Child\` { tcId: $this_connectOrCreate_children_param0 })
+            ON CREATE SET
+                    this_connectOrCreate_children_this1.tcId = $this_connectOrCreate_children_param1
+            MERGE (this_connectOrCreate_children_this1)-[this_connectOrCreate_children_this0:\`HAS\`]->(this)
+            	RETURN COUNT(*)
+            }
+            	WITH this
+            CALL {
+            	WITH this
+            	CALL apoc.util.validate(NOT(ANY(r IN [\\"upstream\\"] WHERE ANY(rr IN $auth.roles WHERE r = rr))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
+            MERGE (this_connectOrCreate_children_this3:\`Child\` { tcId: $this_connectOrCreate_children_param2 })
+            ON CREATE SET
+                    this_connectOrCreate_children_this3.tcId = $this_connectOrCreate_children_param3
+            MERGE (this_connectOrCreate_children_this3)-[this_connectOrCreate_children_this2:\`HAS\`]->(this)
+            	RETURN COUNT(*)
+            }
+            RETURN 'Query cannot conclude with CALL'"
```

Comment:
Before, this statement will only have a single CALL block wrapping the whole thing, meaning that there won't be separation between the second `MERGE` and the second `CALL apoc.util.validate`